File: Problem_Set_Solutions/Week6/sentimental-mario-more/mario.py
from cs50 import get_int  # import library to use get_int


def main():
    while True:
        height = get_int("Height: ")  # ask for height
        if height >= 1 and height <= 8:  # ensure height is between 1 and 8 inclusive
            break
    pyramid(height)  # print the pyramid to user's height


def pyramid(n):
    for i in range(1, n+1):  # from 1 to height inclusive
        print(" "*(n-i), end="")  # print n-i spaces
        print("#"*i, end="")  # print i hashtags
        print("  ", end="")  # print 2 spaces
        print("#"*(i))  # print i more hashtags


main()  # call main

File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Ask the user for the height of the pyramid, ensuring it is an integer more than 0.
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // left side of pyramid
    for (int i = 0; i < n; i++) // for each line
    {
        for (int j = 0; j < (n - i - 1); j++) // print n-i-1 spaces
        {
            printf(" ");
        }
        for (int k = 0; k < i + 1; k++) // print i+1 hashtags
        {
            printf("#");
        }
        printf("  ");
        for (int k = 0; k < i + 1; k++) // print i+1 hashtags
        {
            printf("#");
        }
        printf("\n");
    }
}
